ve --new-file hk_classes-0.8.3/hk_classes/hk_actionquery.cpp hk_classes-0.8.3.new/hk_classes/hk_actionquery.cpp
--- hk_classes-0.8.3/hk_classes/hk_actionquery.cpp      2006-04-13 20:02:01.000000000 +0200
+++ hk_classes-0.8.3.new/hk_classes/hk_actionquery.cpp  2008-04-22 13:38:01.000000000 +0200
@@ -12,6 +12,7 @@
 #include "hk_actionquery.h"
 #include "hk_database.h"

+#include <string.h>
 #include <time.h>

 class hk_actionqueryprivate
diff -u --recursive --new-file hk_classes-0.8.3/hk_classes/hk_button.cpp hk_classes-0.8.3.new/hk_classes/hk_button.cpp
--- hk_classes-0.8.3/hk_classes/hk_button.cpp   2006-07-09 16:33:28.000000000 +0200
+++ hk_classes-0.8.3.new/hk_classes/hk_button.cpp       2008-04-22 13:40:55.000000000 +0200
@@ -17,7 +17,7 @@
 #include "hk_actionquery.h"
 #include "hk_interpreter.h"

-
+#include <string.h>

 class hk_buttonprivate
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_classes/hk_column.cpp hk_classes-0.8.3.new/hk_classes/hk_column.cpp
--- hk_classes-0.8.3/hk_classes/hk_column.cpp   2006-10-02 19:54:17.000000000 +0200
+++ hk_classes-0.8.3.new/hk_classes/hk_column.cpp       2008-04-22 13:42:05.000000000 +0200
@@ -14,6 +14,7 @@
 #include "hk_dsdatavisible.h"
 #include "hk_datetime.h"
 #include <stdio.h>
+#include <string.h>

 hk_column::hk_column(hk_datasource* ds, const hk_string& lTRUE,const hk_string& lFALSE)
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_classes/hk_datasource.cpp hk_classes-0.8.3.new/hk_classes/hk_datasource.cpp
--- hk_classes-0.8.3/hk_classes/hk_datasource.cpp       2006-10-29 20:38:25.000000000 +0100
+++ hk_classes-0.8.3.new/hk_classes/hk_datasource.cpp   2008-04-22 13:43:11.000000000 +0200
@@ -21,7 +21,7 @@
 #include <list>
 #include <stdlib.h>
 #include <time.h>
-
+#include <string.h>

  voidfunction* hk_datasource::p_enablefunction=NULL;
  long hk_datasource::p_enablefunctioncounter=0;
diff -u --recursive --new-file hk_classes-0.8.3/hk_classes/hk_dsimage.cpp hk_classes-0.8.3.new/hk_classes/hk_dsimage.cpp
--- hk_classes-0.8.3/hk_classes/hk_dsimage.cpp  2006-05-20 22:07:19.000000000 +0200
+++ hk_classes-0.8.3.new/hk_classes/hk_dsimage.cpp      2008-04-22 13:44:57.000000000 +0200
@@ -12,6 +12,7 @@
 #include "hk_form.h"
 #include "hk_url.h"

+#include <string.h>

 hk_dsimage::imagetypelisttype  hk_dsimage::p_imagetypefunctions;

diff -u --recursive --new-file hk_classes-0.8.3/hk_classes/hk_reportsection.cpp hk_classes-0.8.3.new/hk_classes/hk_reportsection.cpp
--- hk_classes-0.8.3/hk_classes/hk_reportsection.cpp    2006-08-06 16:37:45.000000000 +0200
+++ hk_classes-0.8.3.new/hk_classes/hk_reportsection.cpp        2008-04-22 13:48:08.000000000 +0200
@@ -19,6 +19,8 @@
 #include "hk_column.h"
 #include "hk_datasource.h"

+#include <string.h>
+
 const  double teiler=10000.0;

 //********************************************
diff -u --recursive --new-file hk_classes-0.8.3/hk_classes/hk_storagedatasource.cpp hk_classes-0.8.3.new/hk_classes/hk_storagedatasource.cpp
--- hk_classes-0.8.3/hk_classes/hk_storagedatasource.cpp        2006-04-13 20:02:05.000000000 +0200
+++ hk_classes-0.8.3.new/hk_classes/hk_storagedatasource.cpp    2008-04-22 13:49:13.000000000 +0200
@@ -10,6 +10,8 @@
 // ****************************************************************************
 #include "hk_storagedatasource.h"

+#include <string.h>
+
 hk_storagedatasource::hk_storagedatasource(hk_database* d,hk_presentation* p):hk_datasource(d,p)
 {
 #ifdef HK_DEBUG
diff -u --recursive --new-file hk_classes-0.8.3/hk_mysqlclasses/hk_mysqlcolumn.cpp hk_classes-0.8.3.new/hk_mysqlclasses/hk_mysqlcolumn.cpp
--- hk_classes-0.8.3/hk_mysqlclasses/hk_mysqlcolumn.cpp 2005-05-16 21:01:50.000000000 +0200
+++ hk_classes-0.8.3.new/hk_mysqlclasses/hk_mysqlcolumn.cpp     2008-04-22 13:52:09.000000000 +0200
@@ -18,6 +18,8 @@
 #include <mysql.h>
 #include <errmsg.h>

+#include <string.h>
+
 hk_mysqlcolumn::hk_mysqlcolumn(hk_mysqldatasource* ds, const hk_string& tTRUE,const hk_string& tFALSE)
 :hk_storagecolumn(ds,tTRUE,tFALSE)
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_mysqlclasses/hk_mysqldatasource.cpp hk_classes-0.8.3.new/hk_mysqlclasses/hk_mysqldatasource.cpp
--- hk_classes-0.8.3/hk_mysqlclasses/hk_mysqldatasource.cpp     2006-06-29 20:10:17.000000000 +0200
+++ hk_classes-0.8.3.new/hk_mysqlclasses/hk_mysqldatasource.cpp 2008-04-22 13:54:17.000000000 +0200
@@ -15,6 +15,7 @@
 #include "hk_mysqlactionquery.h"
 #include <exception>
 #include <new>
+#include <string.h>

 hk_mysqldatasource::hk_mysqldatasource(hk_mysqldatabase* d,hk_presentation* p):hk_storagedatasource(d,p)
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_mysqlclasses/hk_mysqltable.cpp hk_classes-0.8.3.new/hk_mysqlclasses/hk_mysqltable.cpp
--- hk_classes-0.8.3/hk_mysqlclasses/hk_mysqltable.cpp  2006-04-24 20:17:30.000000000 +0200
+++ hk_classes-0.8.3.new/hk_mysqlclasses/hk_mysqltable.cpp      2008-04-22 13:55:09.000000000 +0200
@@ -12,6 +12,9 @@
 #include "hk_mysqltable.h"
 #include "hk_mysqldatabase.h"
 #include "hk_actionquery.h"
+
+#include <string.h>
+
 hk_mysqltable::hk_mysqltable(hk_mysqldatabase* db,hk_presentation* p):hk_mysqldatasource(db,p)

 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_odbcclasses/hk_odbccolumn.cpp hk_classes-0.8.3.new/hk_odbcclasses/hk_odbccolumn.cpp
--- hk_classes-0.8.3/hk_odbcclasses/hk_odbccolumn.cpp   2005-01-26 21:33:41.000000000 +0100
+++ hk_classes-0.8.3.new/hk_odbcclasses/hk_odbccolumn.cpp       2008-04-22 14:01:57.000000000 +0200
@@ -18,6 +18,8 @@
 #include <sql.h>
 //#include <errmsg.h>

+#include <string.h>
+
 hk_odbccolumn::hk_odbccolumn(hk_odbcdatasource* ds, const hk_string& tTRUE,const hk_string& tFALSE)
 :hk_storagecolumn(ds,tTRUE,tFALSE)
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_odbcclasses/hk_odbcdatasource.cpp hk_classes-0.8.3.new/hk_odbcclasses/hk_odbcdatasource.cpp
--- hk_classes-0.8.3/hk_odbcclasses/hk_odbcdatasource.cpp       2006-06-29 20:10:17.000000000 +0200
+++ hk_classes-0.8.3.new/hk_odbcclasses/hk_odbcdatasource.cpp   2008-04-22 14:00:49.000000000 +0200
@@ -18,6 +18,8 @@
 #include <exception>
 #include <new>

+#include <string.h>
+
 hk_odbcdatasource::hk_odbcdatasource(hk_odbcdatabase* d,hk_presentation* p):hk_storagedatasource(d,p)
 {
 #ifdef HK_DEBUG
diff -u --recursive --new-file hk_classes-0.8.3/hk_odbcclasses/hk_odbctable.cpp hk_classes-0.8.3.new/hk_odbcclasses/hk_odbctable.cpp
--- hk_classes-0.8.3/hk_odbcclasses/hk_odbctable.cpp    2006-10-02 19:54:17.000000000 +0200
+++ hk_classes-0.8.3.new/hk_odbcclasses/hk_odbctable.cpp        2008-04-22 14:02:48.000000000 +0200
@@ -16,6 +16,8 @@
 #include <sqltypes.h>
 #include <sqlext.h>

+#include <string.h>
+
 hk_odbctable::hk_odbctable(hk_odbcdatabase* db,hk_presentation* p):hk_odbcdatasource(db,p)

 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_postgresclasses/hk_postgresqlcolumn.cpp hk_classes-0.8.3.new/hk_postgresclasses/hk_postgresqlcolumn.cpp
--- hk_classes-0.8.3/hk_postgresclasses/hk_postgresqlcolumn.cpp 2006-10-02 15:52:53.000000000 +0200
+++ hk_classes-0.8.3.new/hk_postgresclasses/hk_postgresqlcolumn.cpp     2008-04-22 13:56:27.000000000 +0200
@@ -12,6 +12,9 @@
 #include "hk_postgresqlcolumn.h"
 #include "hk_postgresqldatasource.h"
 #include "hk_postgresqldatabase.h"
+
+#include <string.h>
+
 hk_postgresqlcolumn::hk_postgresqlcolumn(hk_postgresqldatasource* ds, const hk_string& tTRUE,const hk_string& tFALSE)
 :hk_storagecolumn(ds,tTRUE,tFALSE)
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_postgresclasses/hk_postgresqldatasource.cpp hk_classes-0.8.3.new/hk_postgresclasses/hk_postgresqldatasource.cpp
--- hk_classes-0.8.3/hk_postgresclasses/hk_postgresqldatasource.cpp     2006-11-27 18:30:13.000000000 +0100
+++ hk_classes-0.8.3.new/hk_postgresclasses/hk_postgresqldatasource.cpp 2008-04-22 13:59:33.000000000 +0200
@@ -17,6 +17,8 @@
 #include <exception>
 #include <new>

+#include <string.h>
+
 hk_postgresqldatasource::hk_postgresqldatasource(hk_postgresqldatabase* d,hk_presentation* p):hk_storagedatasource(d,p)
 {
 #ifdef HK_DEBUG
diff -u --recursive --new-file hk_classes-0.8.3/hk_sqlite3classes/hk_sqlite3column.cpp hk_classes-0.8.3.new/hk_sqlite3classes/hk_sqlite3column.cpp
--- hk_classes-0.8.3/hk_sqlite3classes/hk_sqlite3column.cpp     2006-01-06 15:04:31.000000000 +0100
+++ hk_classes-0.8.3.new/hk_sqlite3classes/hk_sqlite3column.cpp 2008-04-22 14:07:09.000000000 +0200
@@ -18,6 +18,8 @@

 #include <sqlite3.h>

+#include <string.h>
+
 hk_sqlite3column::hk_sqlite3column(hk_sqlite3datasource* ds, const hk_string& tTRUE,const hk_string& tFALSE)
 :hk_storagecolumn(ds,tTRUE,tFALSE)
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_sqlite3classes/hk_sqlite3datasource.cpp hk_classes-0.8.3.new/hk_sqlite3classes/hk_sqlite3datasource.cpp
--- hk_classes-0.8.3/hk_sqlite3classes/hk_sqlite3datasource.cpp 2006-06-29 20:10:18.000000000 +0200
+++ hk_classes-0.8.3.new/hk_sqlite3classes/hk_sqlite3datasource.cpp     2008-04-22 14:07:55.000000000 +0200
@@ -17,6 +17,8 @@
 #include <exception>
 #include <new>

+#include <string.h>
+
 hk_sqlite3datasource::hk_sqlite3datasource(hk_sqlite3database* d,hk_presentation* p):hk_storagedatasource(d,p)
 {
 #ifdef HK_DEBUG
diff -u --recursive --new-file hk_classes-0.8.3/hk_sqliteclasses/hk_sqlitecolumn.cpp hk_classes-0.8.3.new/hk_sqliteclasses/hk_sqlitecolumn.cpp
--- hk_classes-0.8.3/hk_sqliteclasses/hk_sqlitecolumn.cpp       2006-01-06 15:04:31.000000000 +0100
+++ hk_classes-0.8.3.new/hk_sqliteclasses/hk_sqlitecolumn.cpp   2008-04-22 14:05:05.000000000 +0200
@@ -18,6 +18,8 @@

 #include <sqlite.h>

+#include <string.h>
+
 hk_sqlitecolumn::hk_sqlitecolumn(hk_sqlitedatasource* ds, const hk_string& tTRUE,const hk_string& tFALSE)
 :hk_storagecolumn(ds,tTRUE,tFALSE)
 {
diff -u --recursive --new-file hk_classes-0.8.3/hk_sqliteclasses/hk_sqlitedatasource.cpp hk_classes-0.8.3.new/hk_sqliteclasses/hk_sqlitedatasource.cpp
--- hk_classes-0.8.3/hk_sqliteclasses/hk_sqlitedatasource.cpp   2006-06-29 20:10:18.000000000 +0200
+++ hk_classes-0.8.3.new/hk_sqliteclasses/hk_sqlitedatasource.cpp       2008-04-22 14:04:09.000000000 +0200
@@ -17,6 +17,8 @@
 #include <exception>
 #include <new>

+#include <string.h>
+
 hk_sqlitedatasource::hk_sqlitedatasource(hk_sqlitedatabase* d,hk_presentation* p):hk_storagedatasource(d,p)
 {
 #ifdef HK_DEBUG

